src/main/java/TaskList.java

```diff
@@ -0,0 +1,91 @@
+import java.util.ArrayList;
+
+public class TaskList {
+
+    private final ArrayList<Task> myTaskList;
+
+    TaskList() {
+        myTaskList = new ArrayList<>();
+    }
+
+    TaskList(ArrayList<Task> taskList) {
+        this.myTaskList = taskList;
+    }
+
+    ArrayList<Task> getMyTaskList() {
+        return myTaskList;
+    }
+
+    int getListLength() {
+        return myTaskList.size();
+    }
+
+    public void addNewTask(Task newTask) {
+        myTaskList.add(newTask);
+    }
+
+    /**
+     * Marks a task as done. Throws an exception if the task index is invalid.
+     *
+     * @param taskIndex index of task to be marked as done.
+     * @return the done task.
+     * @throws DukeException if index is invalid.
+     */
+    public Task doneTask(int taskIndex) throws DukeException {
+        try {
+            Task temp = myTaskList.get(taskIndex - 1);
+            temp.markAsDone();
+            return temp;
+        } catch (IndexOutOfBoundsException e) {
+            throw new InvalidTaskIndexException("done");
+        }
+    }
+
+    /**
+     * Deletes a task from the list. Throws an exception if the task index is invalid.
+     *
+     * @param taskIndex index of task to be deleted.
+     * @return the deleted task.
+     * @throws DukeException if index is invalid.
+     */
+    public Task deleteTask(int taskIndex) throws DukeException {
+        try {
+            Task temp = myTaskList.get(taskIndex - 1);
+            myTaskList.remove(temp);
+            return temp;
+        } catch (IndexOutOfBoundsException e) {
+            throw new InvalidTaskIndexException("delete");
+        }
+    }
+
+    /**
+     * Finds all the tasks with description that contains the input.
+     *
+     * @param input the required keyword to be searched for.
+     * @return a string describing the list of tasks that contains the input in their description.
+     */
+
```

Comment:
I believe that Coding Standard stated that there should be no spacing between JavaDoc and the method declaration. Perhaps if you could take some time to check it?